include/boost/math/special_functions/chebyshev_transform.hpp

```diff
@@ -0,0 +1,167 @@
+//  (C) Copyright Nick Thompson 2017.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_SPECIAL_CHEBYSHEV_TRANSFORM_HPP
+#define BOOST_MATH_SPECIAL_CHEBYSHEV_TRANSFORM_HPP
+#include <cmath>
+#include <typeinfo>
+#include <type_traits>
+#include <fftw3.h>
+#include <boost/math/constants/constants.hpp>
+#include <boost/math/special_functions/chebyshev.hpp>
+
+namespace boost { namespace math {
+
+template<class Real>
+class chebyshev_transform
+{
+public:
+    template<class F>
+    chebyshev_transform(const F& f, Real a, Real b, Real tol=500*std::numeric_limits<Real>::epsilon()): m_a(a), m_b(b)
```

Comment:
Nick, I think the simplest thing to do here is to add a final enable_if argument:

`chebyshev_transform(const F& f, Real a, Real b, Real tol=500*std::numeric_limits<Real>::epsilon(), typename std::enable_if<std::is_same<double, Real>::value>::type* = nullptr)`

Then this constructor can only be called if type Real is a double, repeat for float and long double and you're done and the if constexpr can go.  It does mean some code duplication, but it's not a huge function so probably not an issue?

Reply:
Yup, will try to get this done!